tests/functional/lib/bash/test_header

```diff
@@ -741,7 +750,7 @@ _parse_platform_spec () {
     local requirement="$1"
 
     local locality='local'
-    local job_runner='background'
+    local job_runner='*'
```

Comment:
This change allows more tests to run under other job submission systems (which was the original intention).

* If a test does *not* specify `REQUIRE_PLATFORM` then it will *only* run on `_local_background`.
* Otherwise it will run on *all* job runners unless `runner:<runner>` is set.